packages/fxa-auth-server/test/local/payments/stripe.js

```diff
@@ -154,4 +155,59 @@ describe('StripeHelper', () => {
     }
     assert.equal(thrown.errno, error.ERRNO.INVALID_PLAN_UPGRADE);
   });
+
+  describe('fetchCustomer', () => {
+    it('fetches an existing customer', async () => {
+      const listResponse = {
+        autoPagingToArray: sinon.fake.resolves([customer1]),
+      };
+      sandbox.stub(stripeHelper.stripe.customers, 'list').returns(listResponse);
+      const result = await stripeHelper.fetchCustomer(
+        'user123',
+        'test@example.com'
+      );
+      assert.deepEqual(result, customer1);
+    });
+
+    it('throws if the customer record has a fxa id mismatch', async () => {
+      const listResponse = {
+        autoPagingToArray: sinon.fake.resolves([customer1]),
+      };
+      sandbox.stub(stripeHelper.stripe.customers, 'list').returns(listResponse);
+      let thrown;
+      try {
+        await stripeHelper.fetchCustomer('user1234', 'test@example.com');
+      } catch (err) {
+        thrown = err;
+      }
+      assert.isObject(thrown);
+      assert.equal(thrown.message, 'A backend service request failed.');
+      assert.equal(
+        thrown.cause().message,
+        'Customer for email: test@example.com in Stripe has mismatched uid'
+      );
+    });
+  });
+
+  describe('findPlayById', () => {
```

Comment:
`s/Play/Plan/` but I might have made this change in my forthcoming branch for #3808 & #3418 since I add tests here